Tutorials/Moving_robot/moving_robot.md

```diff
@@ -0,0 +1,139 @@
+
+```xml
+<plugin
+    filename="libignition-gazebo-diff-drive-system.so"
+    name="ignition::gazebo::systems::DiffDrive">
+    <left_joint>left_wheel_joint</left_joint>
+    <right_joint>right_wheel_joint</right_joint>
+    <wheel_separation>1.2</wheel_separation>
+    <wheel_radius>0.4</wheel_radius>
+    <odom_publish_frequency>1</odom_publish_frequency>
+</plugin>
+```
+
+#### Breaking the code down
+
+The `<plugin>` tag has two attributes, `filename` which takes the library name, `name` takes the name of the plugin. In the `<left_joint>` and `<right_joint>` tags we define the joints which connect the left and right wheel with the body of the robot, in our case `left_wheel_joint` and `right_wheel_joint`. `<wheel_separation>` takes the distance between the two wheels in our robot, We have our `left_wheel` at 0.6 and `right-wheel` -0.6 in y-axis with respect to the `chassis` so the `wheel_separation` is 1.2 . `<wheel_radius>` takes the radius of the wheel which was defined in the `<radius>` tag. `<odom_publish_frequency>` set the frequency by which our car will accept the moving commands it is in `HZ` unit
+
+## Messages and Topics
+
+One of the main concepts in communication in Ignition is "Topics". A topic is simply a name for grouping a specific set of messages or a particular service. The communication happens between two nodes through topics. All the nodes in the network can act as publishers, subscribers, provide services and request services. A publisher is a node that produces information and a subscriber is a node that consumes information.<br/>
+In our case we will send a message to topic name `/model/vehicle_blue/cmd_vel` and our robot subscribe to this topic to get command that make the robot move. Let's do it<br/> 
+Launch the car world `ign gazebo car_world.sdf`
+In another terminal let's send a message to to our car.<br/>
+`ign topic -t "/model/vehicle_blue/cmd_vel" -m ignition.msgs.Twist -p "linear: {x: 0.5}, angular: {z: 0.05}"`
+Now you should have your robot moving in the simulation **Note** don't forget to press the play in the simulation. The command is simple we need to specify topic to publish to after the `-t` option and after the `-m` we specify the message type. Our robot expect message of type `Twist` which consists of two components `linear` and `angular`. after the `-p` option we specify the content(value) of the message. for linear speed `x: 0.5` for angular speed `z: 0.05`.
+For more information about topics and messages in ignition check Transport lib [tutorials](igniton transport)
+
+## Moving the robot using keyboard
+
+Instead of making our robot move in a specific speed we will control our robot using two plugins
+
+### Keystrokes
+
+key strokes is an `ign-gui` plugin catches keyboard's keystrokes and send them on default topic `/gazebo/keyboard/keypress`. Add the following code between the `<world>` tags.
```

Comment:
I think we should call the plugin by a consistent name so the user doesn't get confused. The plugin itself is currently called `KeyPlugin` but here you call it `key strokes`. How about always using the same name.

Also, I'd suggest calling the plugin something like `KeyPublisher`, so that it's clear what it does. We've been keeping the word "plugin" out of plugin names because it's a bit redundant.

What do you think?

---

Also, since that will be a GUI plugin, we should tell the user to put it inside the `<gui>` tag.